plugins/BEdita/Core/src/Model/Table/ObjectsTable.php

```diff
@@ -275,4 +277,61 @@ public function getId($uname)
 
         return $result['id'];
     }
+
+    /**
+     * Finder for objects having a certain `ancestor` on the tree.
+     *
+     * @param Query $query  Query object instance.
+     * @param array $options Id or unique name of ancestor
+     * @return \Cake\ORM\Query
+     */
+    protected function findAncestor(Query $query, array $options)
+    {
+        return $this->treeDescendants($query, Hash::get($options, '0'), false);
+    }
+
+    /**
+     * Finder for objects having a certain `parent` on the tree.
+     *
+     * @param Query $query Query object instance.
+     * @param array $options Id or unique name of ancestor
+     * @return \Cake\ORM\Query
+     */
+    protected function findParent(Query $query, array $options)
+    {
+        return $this->treeDescendants($query, Hash::get($options, '0'), true);
+    }
+
+    /**
+     * Create query on Trees for `findParent` and `findAncestor`
+     *
+     * @param Query $query Query object instance.
+     * @param int|string $id Parent id or unique name
+     * @param bool $direct Flag to look for direct descendants only, a.k.a. children
+     * @return \Cake\ORM\Query
+     */
+    protected function treeDescendants(Query $query, $id, $direct = true)
+    {
+        $query = $query->join([
+            'table' => 'trees',
+            'alias' => 'Trees',
+            'type' => 'INNER',
+            'conditions' => 'Trees.object_id = ' . $this->aliasField('id'),
+        ]);
```

Comment:
[super minor]: The `order()` could be moved here since is always the same

Reply:
no more necessary, after new refactor